lib/src/bootloader.rs

```diff
@@ -26,3 +26,113 @@ pub(crate) fn install_via_bootupd(
         .verbose()
         .run()
 }
+
+#[context("Installing bootloader using zipl")]
+pub(crate) fn install_via_zipl(
+    device: &Utf8Path,
+    boot_uuid: &str,
+) -> Result<()> {
+
+    // Run findmnt to find the boot partition
+
+    #[derive(serde::Deserialize)]
+    struct Filesystem {
+        source: String,
+        target: String,
+    }
+
+    #[derive(serde::Deserialize)]
+    struct Findmnt {
+        filesystems: Vec<Filesystem>,
+    }
+
+    let findmnt_desc = format!("Running findmnt to identify the boot partition with UUID={boot_uuid}");
+    let findmnt_task = Task::new(findmnt_desc, "findmnt");
+    let findmnt_args = ["--json", "--output", "TARGET,SOURCE,FSTYPE",
+                        "--uniq", "--first-only", "--nofsroot",
+                        "--source", &(format!("UUID={boot_uuid}"))];
+    let findmnt_out  = findmnt_task.args(findmnt_args).quiet().read()?;
+    let findmnt_json = serde_json::from_str::<Findmnt>(&findmnt_out).context("parsing findmnt output")?;
+
+    let boot_fs = findmnt_json.filesystems.get(0).context("finding boot filesystem")?;
+    let boot_dev = Utf8Path::new(&boot_fs.source);
+    let boot_dir = Utf8Path::new(&boot_fs.target);
+
+    // Run sfdisk to determine the boot partition offset
+
+    #[derive(serde::Deserialize)]
+    struct Partition {
+        node:  String,
+        start: i64,
+    }
+
+    #[derive(serde::Deserialize)]
+    struct PartitionTable {
+        sectorsize: i64,
+        partitions: Vec<Partition>,
+    }
+
+    #[derive(serde::Deserialize)]
+    struct Sfdisk {
+        partitiontable: PartitionTable
+    }
+
+    let sfdisk_desc = format!("running sfdisk to determine the partition offset of {boot_dev}");
+    let sfdisk_task = Task::new(sfdisk_desc, "sfdisk");
+    let sfdisk_args = ["--json", device.as_str()];
+    let sfdisk_out  = sfdisk_task.args(sfdisk_args).quiet().read()?;
+    let sfdisk_json = serde_json::from_str::<Sfdisk>(&sfdisk_out).context("parsing findmnt output")?;
+
+    let block_size  = sfdisk_json.partitiontable.sectorsize;
+    let partitions  = sfdisk_json.partitiontable.partitions;
+    let boot_part   = partitions.iter().find(|p| p.node == boot_dev).context("finding boot partition")?;
+    let boot_offset = boot_part.start;
+
+    // Find the default BLS conf under /boot/loader/entries
+
+    let bls_dir = boot_dir.join("boot/loader/entries");
```

Comment:
I think this code looks like it originally came from coreos-installer? This is fine for now, but it is a bit of a duplication as ostree already has a BLS parser too...and I think here because (unlike coreos-installer) we are hard linked to ostree today we could actually reference the `ostree::Deployment` object which has an associated boot loader config object we could use.

Reply:
Actually, I referenced the logic in the zipl source code.
https://github.com/ibm-s390-linux/s390-tools/blob/b6c30404bcceca1f7328abab4df2be8535667564/zipl/src/scan.c#L1113

Due to my lack of knowledge about ostree, I'd like to use the Rust code here for now, and will add a TODO comment.